evalml/preprocessing/utils.py

```diff
@@ -42,6 +43,53 @@ def load_data(path, index, target, n_rows=None, drop=None, verbose=True, **kwarg
     return infer_feature_types(X), infer_feature_types(y)
 
 
+def split_multiseries_data(X, y, series_id, time_index, **kwargs):
+    """Split stacked multiseries data into train and test sets. Unstacked data can use `split_data`.
+
+    Args:
+        X (pd.DataFrame): The input training data of shape [n_samples*n_series, n_features].
+        y (pd.Series): The target training targets of length [n_samples*n_series].
+        series_id (str): Name of column containing series id.
+        time_index (str): Name of column containing time index.
+        **kwargs: Additional keyword arguments to pass to the split_data function.
+
+    Returns:
+        pd.DataFrame, pd.DataFrame, pd.Series, pd.Series: Feature and target data each split into train and test sets.
+    """
+    X_unstacked, y_unstacked = unstack_multiseries(
+        X,
+        y,
+        series_id,
+        time_index,
+        y.name,
+    )
+    (
+        X_train_unstacked,
+        X_holdout_unstacked,
+        y_train_unstacked,
+        y_holdout_unstacked,
+    ) = split_data(
+        X_unstacked, y_unstacked, problem_type="time series regression", **kwargs
+    )
+
+    X_train = restack_X(X_train_unstacked, series_id, time_index)
+    X_holdout = restack_X(X_holdout_unstacked, series_id, time_index)
+    y_train = stack_data(y_train_unstacked)
```

Comment:
would it make sense to rename either the X or y function to be named the same? i.e (restack_X, restack_y) or (stack_X, stack_y)?

Reply:
I understand the desire but am having trouble settling on a good pair of names. It doesn't make sense to rename `stack_data` to either `stack_y` or `restack_y`, because `stack_data` is meant to stack/restack any single stacked column worth of data, including within `restack_X`, where it's called for each original column. I could see renaming `restack_X` to `stack_X` for a bit more consistency, since the `re` isn't a requirement, I'll try that.